File: editor/backend/releng/de.jabc.cinco.meta.core.parent/de.jabc.cinco.meta.plugin.pyro/frontend/app/lib/src/pages/shared/navigation/navigation_component.dart
import 'package:angular/angular.dart';
import 'dart:html';
import 'package:angular_router/angular_router.dart';

import '../../main/routes.dart';
import '../../main/route_paths.dart';
import '../../../routes.dart' as top_routes;

import '../../../service/style_service.dart';
import '../../../pages/profile/profile_image/profile_image_component.dart';
import '../../../model/core.dart';

@Component(
  selector: 'navigation',
  templateUrl: 'navigation_component.html',
  directives: const [coreDirectives, routerDirectives, ProfileImageComponent],
  styleUrls: const ['navigation_component.css'],
  providers: const [],
  exports: const [top_routes.RoutePaths]
)
class NavigationComponent {

  final Router _router;
  final StyleService _styleService;
  
  @Input("user")
  PyroUser currentUser;
  
  NavigationComponent(this._router, this._styleService) {
  }
  
  bool get isAdmin => currentUser != null && currentUser.systemRoles.contains(PyroSystemRole.ADMIN);
  
  PyroStyle get style => _styleService.style;
}

File: services/main/frontend/src/app/features/internal/components/navigation/navigation.component.ts
import { Component, OnInit } from '@angular/core';
import { AppStoreService } from '../../../../core/services/stores/app-store.service';
import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
import { User } from '../../../../core/models/user';

@UntilDestroy()
@Component({
  selector: 'cc-navigation',
  templateUrl: './navigation.component.html',
  styleUrls: ['./navigation.component.css']
})
export class NavigationComponent implements OnInit {

  public user: User;

  constructor(private appStore: AppStoreService) {
  }

  ngOnInit(): void {
    this.appStore.user$.pipe(untilDestroyed(this)).subscribe({
      next: user => this.user = user
    });
  }
}
